src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -655,6 +658,11 @@ static int do_map(int argc, const char *argv[], Config *cfg)
     read_only = 1;
   }
 
+  change_admin_socket(index);
```

Comment:
We discussed this earlier too. After this change the users lost a possibility to change admin socket path via config. I like the idea to have "nbdX" in the socket name so actually I don't mind. But as far as I recall Jason did not like this idea and suggested to extend config implementation to support user defined (client specific) meta variables so we could use e.g. `{nbd}` meta variable in `admin_socket` config option.

@dillaman what do you think?

Reply:
Indeed -- I don't want to take a step backwards and have rbd-nbd automagically override configuration settings.  I think a plugable set of metavariables would be helpful in other scenarios as well (but that shouldn't be part of this PR).

Reply:
Actually, I think allowing users to override `admin_socket` is very important. Because now they can map images with a command like below:
```
rbd-nbd --admin-socket=/tmp/${image_name}.asok map ${image_name}
```
And would lose this possibility after your change. You could change your code to change admin socket only if it is not specified, then I would not mind, but I would like to see what Jason thinks.

Reply:
@trociny I don't like the user to specify --admin-socket, because I think it is hard to specify it:
1) As you said, ${image_name}, but the we allow the user to map an image more than once in same client, and it is useful in our apps. So ${image_name} could not be used. I also consider pid, but also not workable, because in "do_map", main process will be forked.
2) I used "nbdX" in our product, and the feedback from our customer is good.

I can understand your concern, you may want to give flexibility. I could remove this commit from this PR.